README.md

```diff
@@ -4,3 +4,218 @@
+the following commands.
+
+```bash
+$ opam init --bare
+$ opam switch create ocaml-base-compiler.4.09.0
+$ eval $(opam env)
+# from the Beluga directory
+$ opam install --deps-only .
+```
+
+For more information about WSL
+  <https://docs.microsoft.com/en-us/windows/wsl/about>
+
+
+Alternatively, if you use Windows 8 or earlier versions, you can build and execute
+Beluga on Windows through Cygwin. The necessary prerequisites can be installed
+using opam for Windows.
+
+
+1. Download opam for Windows' graphical installer OCaml32/64.exe
+  <https://fdopen.github.io/opam-repository-mingw/installation/>
+
+2. Run OCaml32/64.exe and step through the installation wizard
+  Note: if you don't already have Cygwin installed,
+  it will be installed for you.
+
+3. Run the following commands from Beluga directory within cygwin terminal.
+
+```bash
+$ opam switch create ocaml-variants.4.09.0+mingw64c
+$ eval $(opam env)
+# from the Beluga directory
+$ opam install --deps-only .
+```
+
+
+For more inforamtion about opam for Windows
+  <https://fdopen.github.io/opam-repository-mingw>
+For more information about Cygwin
+  <http://www.cygwin.com/>
+
+## Building
+
+
+Compile by running "make" from the Beluga directory.
+```bash
+$ make
+```
+
+You can now run Beluga programs with the newly
+"beluga" executable in the "bin" directory
+
+```bash
+$ ./bin/beluga path/to/program.bel
+```
+
+
+#### Compilation Tips
+
+
+Running `make clean` will clean the directory of compilation results
+
+
+#### Troubleshooting
+
+
+Attempting to build Beluga before installing the prequisites may cause
+libraries to be only partially updated. You may recieve a new error message
+once you've set up the prereq packages.
+
+```bash
+Error: The files [path/to/file]
+        and [path/to/file]
+        make inconsistent assumptions over interface [library name]
+```
+Try cleaning the Beluga directory.
+
+```bash
+$ make clean all
+```
+
+#### Configuration
+
+
+You will need Ruby 1.9 for running the script.
+
+For MAC OS:
+
+The default version of ruby installed on Mac OS is usually not Ruby 1.9, but
+Ruby 1.8. To install the newest version of Ruby, we recommend installing first
+RVM, the “Ruby Version Manager. This will make installing Ruby 1.9 easy.
+In your home directory, you can say:
+
+```bash
+$ bash < <(curl -s https://rvm.beginrescueend.com/install/rvm)
+```
+
+rvm is then installed in .rvm/bin.
+
+Then you can say:
+
+```
+$ .rvm/bin/rvm install 1.9.3
+```
+
+Note, you will want to add .rvm/bin to your .bashrc directory, so you will be
+able to find rvm without giving the full path.
+
+## Interactive Mode
+
+Beluga has a (legacy) interactive mode for Emacs. Check docs [here](https://github.com/Beluga-lang/Beluga/blob/master/interactive-mode-testing.md).
+We strongly advise using [Harpoon](https://github.com/Beluga-lang/Beluga/blob/master/harpoon.md). A detailed list of commands and tactics is listed [here](https://beluga-lang.readthedocs.io/en/latest/harpoon/index.html).
```

Comment:
This `harpoon.md` is outdated. I think both link should point readthedocs, or change both the first and second link into something like
```
[Harpoon](https://beluga-lang.readthedocs.io/en/latest/harpoon/index.html)
[here](`https://beluga-lang.readthedocs.io/en/latest/harpoon/interactive-reference.html`)
```